spock-specs/src/test/groovy/org/spockframework/mock/runtime/MockMakerRegistrySpec.groovy

```diff
@@ -0,0 +1,162 @@
+package org.spockframework.mock.runtime
+
+import org.spockframework.mock.CannotCreateMockException
+import org.spockframework.mock.IMockObject
+import org.spockframework.util.InternalSpockError
+import spock.lang.Specification
+import spock.mock.MockMakerId
+
+class MockMakerRegistrySpec extends Specification {
+
+  def "No MockMaker"() {
+    when:
+    new MockMakerRegistry([], null)
+    then:
+    thrown(InternalSpockError)
+  }
+
+  def "Invalid id: null"() {
+    given:
+    def maker = Mock(IMockMaker)
+    when:
+    new MockMakerRegistry([maker], null)
+    then:
+    def ex = thrown(IllegalStateException)
+    ex.message == "The IMockMaker 'Mock for type 'IMockMaker' named 'maker'' has an invalid ID: null"
+  }
+
+  def "Invalid id: incorrect suffix"() {
+    given:
+    def maker = Mock(IMockMaker)
+    maker.getId() >> "sample"
+    when:
+    new MockMakerRegistry([maker], null)
+    then:
+    def ex = thrown(IllegalStateException)
+    ex.message == "The IMockMaker 'Mock for type 'IMockMaker' named 'maker'' has an invalid ID: sample"
+  }
+
+  def "Invalid priority"() {
+    given:
+    def maker = Mock(IMockMaker)
+    maker.getId() >> MockMakerId.dynamicProxy
+    maker.getPriority() >> -1
+    when:
+    new MockMakerRegistry([maker], null)
+    then:
+    def ex = thrown(IllegalStateException)
+    ex.message == "The IMockMaker 'dynamic-proxy-mock-maker' has an invalid priority: -1"
+  }
+
+  def "Duplicate mock makers"() {
+    given:
+    def maker1 = Mock(IMockMaker)
+    def maker2 = Mock(IMockMaker)
+    maker1.getId() >> MockMakerId.dynamicProxy
+    maker2.getId() >> MockMakerId.dynamicProxy
+    when:
+    new MockMakerRegistry([maker1, maker2], null)
+    then:
+    def ex = thrown(IllegalStateException)
+    ex.message == "Duplicated IMockMaker instances with the same ID found: dynamic-proxy-mock-maker"
+  }
+
+  def "MockMaker sorting"() {
+    given:
+    def proxy = new DynamicProxyMockMaker()
+    def mockSamePriorityProxy = Mock(IMockMaker)
+    mockSamePriorityProxy.getId() >> "sameprio-mock-maker"
+    def byteBuddy = new ByteBuddyMockMaker()
+    def mockMockMaker = Mock(IMockMaker)
+    mockMockMaker.getId() >> "sample-mock-maker"
+    mockMockMaker.getPriority() >> 5
+    def cglib = new CgLibMockMaker()
+    when:
+    def r = new MockMakerRegistry([mockSamePriorityProxy, byteBuddy, cglib, mockMockMaker, proxy], null)
+    then:
+    r.getMakerList() == [proxy, mockSamePriorityProxy, mockMockMaker, byteBuddy, cglib]
+  }
+
+  def "MockMaker sorting with preferred"() {
+    given:
+    def proxy = new DynamicProxyMockMaker()
+    def mockSamePriorityProxy = Mock(IMockMaker)
+    mockSamePriorityProxy.getId() >> "sameprio-mock-maker"
+    def byteBuddy = new ByteBuddyMockMaker()
+    def mockMockMaker = Mock(IMockMaker)
+    mockMockMaker.getId() >> "sample-mock-maker"
```

Comment:
:x: Make the name match the id, then it is easier to understand.

Reply:
Done.